workspaces/liferay-customer-workspace/client-extensions/liferay-customer-etc-spring-boot/src/main/java/com/liferay/customer/AccountBusinessEventsRestController.java

```diff
@@ -79,6 +93,77 @@ public ResponseEntity<String> post(
 		}
 	}
 
+	private void _checkPermission(Jwt jwt, String externalReferenceCode)
+		throws Exception {
+
+		AccountResource accountResource = AccountResource.builder(
+		).header(
+			HttpHeaders.AUTHORIZATION, "Bearer " + jwt.getTokenValue()
+		).endpoint(
+			new URL(getWebClientBaseURL())
+		).build();
+
+		Account account = accountResource.getAccountByExternalReferenceCode(
+			externalReferenceCode);
+
+		UserAccountResource userAccountResource = UserAccountResource.builder(
+		).header(
+			HttpHeaders.AUTHORIZATION, "Bearer " + jwt.getTokenValue()
+		).endpoint(
+			new URL(getWebClientBaseURL())
+		).build();
+
+		UserAccount userAccount = userAccountResource.getMyUserAccount();
+
+		JSONObject userJSONObject = new JSONObject(userAccount.toString());
```

Comment:
Instead of converting to a JSONObject here, can you use the methods it already has? (getAccountBriefs)